lib/manager/git-submodules/update.ts

```diff
@@ -21,3 +35,61 @@ export default async function updateDependency({
     return null;
   }
 }
+
+function getGitEnvironmentVariables(): NodeJS.ProcessEnv {
+  let environmentVariables: NodeJS.ProcessEnv = {};
+
+  // hard-coded logic to use authentication for github.com based on the githubToken for api.github.com
+  const githubToken = find({
+    hostType: PlatformId.Github,
+    url: 'https://api.github.com/',
+  });
+
+  if (githubToken?.token) {
+    environmentVariables = getGitAuthenticatedEnvironmentVariables(
+      'https://github.com/',
+      githubToken
+    );
+  }
+
+  // get extra host rules for other git-based Go Module hosts
+  const hostRules = getAll() || [];
```

Comment:
We shouldn't use all hostrules. Instead we should read all submodule urls and check for matching hostrules only.